internal/services/mobilenetwork/helpers.go

```diff
@@ -0,0 +1,44 @@
+package mobilenetwork
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+// a workaround for that some child resources may still exist for seconds before it fully deleted.
+// it will cause the error "Can not delete resource before nested resources are deleted."
+func resourceMobileNetworkChildWaitForDeletion(ctx context.Context, id string, getFunction func() (*http.Response, error)) error {
+	deadline, _ := ctx.Deadline()
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending: []string{"200"},
+		Target:  []string{"404"},
+		Refresh: func() (interface{}, string, error) {
+			resp, err := getFunction()
+			if err != nil {
+				if response.WasNotFound(resp) {
+					return resp, strconv.Itoa(resp.StatusCode), nil
+				}
+
+				return nil, strconv.Itoa(resp.StatusCode), fmt.Errorf("polling for %s: %+v", id, err)
+			}
+
+			return resp, strconv.Itoa(resp.StatusCode), nil
+		},
+		MinTimeout:                10 * time.Second,
+		PollInterval:              10 * time.Second,
+		ContinuousTargetOccurence: 6,
+		Timeout:                   time.Until(deadline),
+	}
+
+	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for deleting %s: %+v", id, err)
```

Comment:
```suggestion
		return fmt.Errorf("waiting for deletion of %s: %+v", id, err)
```

Reply:
Can we let this be changed with #20334 to avoid merge conflicts?..

Reply:
Yes that makes sense, thanks @ziyeqf!